### PR TITLE
CORS changes

### PR DESCRIPTION
**Pull request by jstavh**
to mina.vote from any

### CODE REVIEW

File: server/src/main.rs [hunk 1]
```diff
@@ -31,7 +31,7 @@ async fn main() -> anyhow::Result<()> {
 
     let cors = CorsLayer::new()
         .allow_methods([Method::GET, Method::POST])
-        .allow_origin(Any);
+        .allow_origin("https://mina.vote/".parse::<HeaderValue>().unwrap());
```
Comment (trevorbernard) [inline server/src/main.rs:34]: This doesn't take into consideration the staging URL. I think development env should allow `Any`

**Review by trevorbernard — CHANGES_REQUESTED** [review]

File: server/src/main.rs (hunk 1 above)
Comment (jstavh) [reply server/src/main.rs:34]: Good point. I'm working on adding this

Comment (trevorbernard) [reply server/src/main.rs:34]: https://docs.rs/tower-http/latest/tower_http/cors/struct.CorsLayer.html#method.allow_origin

File: server/src/main.rs [hunk 2]
```diff
@@ -29,9 +29,16 @@ async fn main() -> anyhow::Result<()> {
         .await
         .context("Error: Could not connect to mainnet database.")?;
 
+    let origins = [
+        "https://mina.vote/".parse::<HeaderValue>().unwrap(),
+        "http://99.79.130.169:8080/".parse::<HeaderValue>().unwrap(),
```
Comment (trevorbernard) [inline server/src/main.rs:34]: I thought we agreed that we would get this value from an ENV variable and default to ANY?

Comment (trevorbernard) [reply server/src/main.rs:34]: @jstavh This should be pulled from an ENV variable to make it more configurable across environments. If we don't set a value or it's empty, it should default to allowing `ANY` maintaining backwards compatibility.

**Review by trevorbernard — CHANGES_REQUESTED** [review]

**Comment by trevorbernard** [comment]
There is also a merge conflict with the Cargo.toml and Cargo.lock file

**Comment by jstavh** [comment]
> There is also a merge conflict with the Cargo.toml and Cargo.lock file

Fixed this

**Review by trevorbernard — APPROVED** [review]

File: server/src/main.rs [hunk 3]
```diff
@@ -29,9 +29,19 @@ async fn main() -> anyhow::Result<()> {
         .await
         .context("Error: Could not connect to mainnet database.")?;
 
+    let allowed_origins = match std::env::var("ALLOWED_ORIGINS") {
+        Ok(val) => {
+            let origin = val.split(",").map(|s| s.trim()).collect::<Vec<&str>>().join(",");
```
Comment (trevorbernard) [inline server/src/main.rs:34]: What happens with ALLOWED_ORIGINS is empty?

**Review by jstavh — COMMENTED** [review]
Made a new commit to address the most recent comment.

**Review by trevorbernard — APPROVED** [review]

**Pull request merged**